clusterloader2/api/types.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"time"
+)
+
+// Config is a structure that represents configuration
+// for a single test scenario.
+type Config struct {
+	// AutomanagedNamespaces is a number of automanaged namespaces.
+	AutomanagedNamespaces int32 `json: automanagedNamespaces`
+	// Steps is a sequence of test steps executed in serial.
+	Steps []Step `json: steps`
+	// TuningSets is a collection of tuning sets that can be used by steps.
+	TuningSets []TuningSet `json: tuningSets`
+}
+
+// Step represents encapsulation of some actions. These actions could be
+// object declarations or measurement usages.
+// Exactly one structure (Phases or Measurements) should be non-empty.
+type Step struct {
+	// Phases is a collection of declarative definitions of objects.
+	// Phases will be executed in parallel.
+	Phases []Phase `json: phases`
+	// Measurements is a collection of parallel measurement calls.
+	Measurements []Measurement `json: measurements`
+}
+
+// Phase is a structure that declaratively defines state of objects.
+// In a given namespace range (or cluster scope if no range is specified)
+// it defines the number and the configuration of managed objects.
+type Phase struct {
+	// NamespaceRange defines the set of namespaces in which objects
+	// should be reconciled.
+	// If null, objects are assumed to be cluster scoped.
+	NamespaceRange *NamespaceRange `json: namespaceRange`
+	// ReplicasPerNamespace is a number of instances of a given object
+	// to exist in each of referenced namespaces.
+	ReplicasPerNamespace int32 `json: replicasPerNamespace`
+	// TuningSet is the name of TuningSet to be used.
+	TuningSet string `json: tuningSet`
+	// Objects defines a set of object definitions.
```

Comment:
Sorry - I know it's my fault, but this sentence sounds strange - sorry for this suggestion.
Maybe just:
"Objects declaratively defines a set of objects."

Reply:
done.
However, I think we want to introduce "object bundle" or something that is not just set of objects. The problem is that we lack a word or name for objects set with assumption that is a package not just a  group of individual objects.
Should I change it back to "object bundle" instead of "set of objects"?

Reply:
So maybe the solution is to call this field "ObjectBundle" (instead of Objects) ?

Reply:
Sounds good to me.